compiler/rustc_codegen_cranelift/src/unsize.rs

```diff
@@ -31,9 +31,7 @@ pub(crate) fn unsized_info<'tcx>(
             // change to the vtable.
             old_info.expect("unsized_info: missing old info for trait upcast")
         }
-        (_, &ty::Dynamic(ref data, ..)) => {
-            crate::vtable::get_vtable(fx, fx.layout_of(source), data.principal())
-        }
+        (_, &ty::Dynamic(ref data, ..)) => crate::vtable::get_vtable(fx, source, data.principal()),
```

Comment:
This needs to stay the same. `get_vtable` and `build_vtable` are not the same method. The former handles caching, while the later only builds the vtable.

Reply:
I intended to change the signature of `get_vtable` method, but got it wrong. Now i've fixed it.